mlos_bench/mlos_bench/launcher.py

```diff
@@ -227,6 +230,20 @@ def _load_config(self,
             global_config["config_path"] = config_path
         return global_config
 
+    def _expand_vars(self, value: Any) -> Any:
```

Comment:
We do this in several places now.  Is it possible for us to consolidate this into `mlos_bench.util` and fixup all the places where we do `$var` expansion to call it there?  (see also comments about recursive calls below)

Reply:
I might do it later. Right now, we use variable expansion in four different contexts and at different stages of the Environment life cycle:
* Global parameters: we do the expansion once at the startup time; both input and output are the same global parameters
* `LocalFileShareEnv`: we substitute `const_args` and tunables (plus the special variable `$PWD`) inside the file paths for upload and download
* `tunable_params_map`: we substitute the _lists_ of covariant groups' names inside the `tunable_params`
* `LocalEnv` and `RemoteEnv`: we pass `shell_env_params` as shell environment variables of the executable script

Note that all these cases use _different_ parameters for the expansions, and the expansion mechanisms are different, too (e.g., string substitution in globals vs. list expansion for the tunables). That makes it hard to have a single unified function for all use cases. I would argue that adding string substitution for the globals covers most of our needs for the flexible configs for the nearest future; later we can rethink the way we handle the dollar variables in our configs and come up with a more elegant solution